lib/cache/repositories/impressions/memory_repository.rb

```diff
@@ -18,13 +18,14 @@ def add(split_name, data)
             end
           end
 
-          def add_bulk(key, bucketing_key, treatments, time)
-            treatments.each do |split_name, treatment|
+          def add_bulk(key, bucketing_key, treatments_with_labels, time)
+            treatments_with_labels.each do |split_name, treatment_with_label|
               add(
                 split_name,
                 'key_name' => key,
                 'bucketing_key' => bucketing_key,
-                'treatment' => treatment,
+                'treatment' => treatment_with_label[:treatment],
+                'label' => treatment_with_label[:label],
```

Comment:
what happen to `treatment_with_label[:label]` if label does not exist? Which could be a possibility.